File: src/main/java/com/kingdom/model/RandomizingOptions.java
package com.kingdom.model;

import java.util.ArrayList;
import java.util.List;

public class RandomizingOptions {
    private boolean oneOfEachCost;
    private boolean oneWithBuy;
    private boolean oneWithActions;
    private boolean defenseForAttack;
    private boolean threeToFiveAlchemy;
    private List<Card> customSelection = new ArrayList<Card>(0);
    private List<Card> excludedCards = new ArrayList<Card>(0);
    private boolean swappingCard;
    private int cardToReplaceIndex;
    private Card cardToReplace;
    private String cardTypeToReplaceWith;

    public boolean isOneOfEachCost() {
        return oneOfEachCost;
    }

    public void setOneOfEachCost(boolean oneOfEachCost) {
        this.oneOfEachCost = oneOfEachCost;
    }

    public boolean isOneWithBuy() {
        return oneWithBuy;
    }

    public void setOneWithBuy(boolean oneWithBuy) {
        this.oneWithBuy = oneWithBuy;
    }

    public boolean isOneWithActions() {
        return oneWithActions;
    }

    public void setOneWithActions(boolean oneWithActions) {
        this.oneWithActions = oneWithActions;
    }

    public boolean isDefenseForAttack() {
        return defenseForAttack;
    }

    public void setDefenseForAttack(boolean defenseForAttack) {
        this.defenseForAttack = defenseForAttack;
    }

    public boolean isThreeToFiveAlchemy() {
        return threeToFiveAlchemy;
    }

    public void setThreeToFiveAlchemy(boolean threeToFiveAlchemy) {
        this.threeToFiveAlchemy = threeToFiveAlchemy;
    }

    public List<Card> getCustomSelection() {
        return customSelection;
    }

    public void setCustomSelection(List<Card> customSelection) {
        this.customSelection = customSelection;
    }

    public List<Card> getExcludedCards() {
        return excludedCards;
    }

    public void setExcludedCards(List<Card> excludedCards) {
        this.excludedCards = excludedCards;
    }

    public boolean isSwappingCard() {
        return swappingCard;
    }

    public void setSwappingCard(boolean swappingCard) {
        this.swappingCard = swappingCard;
    }

    public int getCardToReplaceIndex() {
        return cardToReplaceIndex;
    }

    public void setCardToReplaceIndex(int cardToReplaceIndex) {
        this.cardToReplaceIndex = cardToReplaceIndex;
    }

    public Card getCardToReplace() {
        return cardToReplace;
    }

    public void setCardToReplace(Card cardToReplace) {
        this.cardToReplace = cardToReplace;
    }

    public String getCardTypeToReplaceWith() {
        return cardTypeToReplaceWith;
    }

    public void setCardTypeToReplaceWith(String cardTypeToReplaceWith) {
        this.cardTypeToReplaceWith = cardTypeToReplaceWith;
    }
}

File: src/main/kotlin/com/kingdom/model/RandomizingOptions.kt
package com.kingdom.model

import com.kingdom.model.cards.*
import java.util.ArrayList

class RandomizingOptions {
    var isOneOfEachCost: Boolean = false
    var isOneWithBuy: Boolean = false
    var isOneWithActions: Boolean = false
    var isDefenseForAttack: Boolean = false
    var customCardSelection: List<Card> = ArrayList(0)
    var numEventsAndLandmarksAndProjectsAndWays: Int = 2
    var customEventSelection: List<Event> = ArrayList(0)
    var customLandmarkSelection: List<Landmark> = ArrayList(0)
    var customProjectSelection: List<Project> = ArrayList(0)
    var customWaySelection: List<Way> = ArrayList(0)
    var customTraitSelection: List<Trait> = ArrayList(0)
    var isIncludeEvents: Boolean = true
    var isIncludeLandmarks: Boolean = true
    var isIncludeProjects: Boolean = true
    var isIncludeWays: Boolean = true
    var isIncludeTraits: Boolean = true
    var excludedCards: List<Card> = ArrayList(0)
    var isSwappingCard: Boolean = false
    var cardToReplaceIndex: Int = 0
    var cardToReplace: Card? = null
}
